cubedash/_pages.py

```diff
@@ -197,6 +197,11 @@ def search_page(
         _model.STORE.index.datasets.search(**query, limit=_HARD_SEARCH_LIMIT + 1),
         key=lambda d: d.center_time,
     )
+
+    for i in range(len(datasets)):
+        if _model.STORE.get_dataset_center_time(datasets[i].id):
+            datasets[i].center_time = _model.STORE.get_dataset_center_time(datasets[i].id)
```

Comment:
This is going to result in hundreds of database queries, run once a time, before the page can be loaded. This is very slow on some environments, such as AWS+RDS where query latency is non-trivial.

(We faced a similar problem previously on the storage page on AWS, which initially had a query per product, which is far fewer queries than these two.)

If we do this, we should combine this loop into one single query, or return the info from the existing search query.